lib/statsd/instrument/client.rb

```diff
@@ -22,10 +22,11 @@ def initialize(
     prefix: nil,
     default_sample_rate: 1,
     default_tags: nil,
-    datagram_builder_class: StatsD::Instrument::StatsDDatagramBuilder
+    implementation: 'datadog',
+    datagram_builder_class: nil
```

Comment:
You could handle the default here instead of in the method implementation

```suggestion
    datagram_builder_class: datagram_builder_class_from_implementation(implementation)
```

Reply:
:+1: